### PR TITLE
Remove toString from Painless user tree nodes

### PR DESCRIPTION
**Pull request by jdconrad**
This removes the custom toString method from the user tree nodes. The current rendition of toString depends on information generated prior to the division of the original AST. These methods should be added back in (https://github.com/elastic/elasticsearch/issues/54116), but in the meantime cause a lot of overhead to maintain as the structure continues to evolve.

This change also moves a bit of data out of the SClass node and into the "environment" as it's only required for the ir tree, not for semantic checking.

Relates: #53702

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java [hunk 1]
```diff
@@ -170,26 +169,21 @@
 public final class Walker extends PainlessParserBaseVisitor<ANode> {
 
     public static SClass buildPainlessTree(ScriptClassInfo mainMethod, String sourceName,
-                                            String sourceText, CompilerSettings settings, PainlessLookup painlessLookup,
-                                            Printer debugStream) {
-        return new Walker(mainMethod, sourceName, sourceText, settings, painlessLookup, debugStream).source;
+                                            String sourceText, CompilerSettings settings, PainlessLookup painlessLookup) {
```
Comment (stu-elastic) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java]: fix spacing

**Review by stu-elastic — APPROVED** [review]
Looks good!

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/antlr/Walker.java]: Done.

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review. Will merge as soon as CI passes.

**Comment by jdconrad** [comment]
@elasticsearchmachine run elasticsearch-ci/default-distro

**Pull request merged**